apps/frontend/src/app/Database/Database.test.ts

```diff
@@ -702,4 +706,47 @@ describe('Database', () => {
       database.weapons.get(database.chars.get('Albedo')?.equippedWeapon)?.key
     ).toEqual('DullBlade')
   })
+
+  describe('Test character talent validation', () => {
```

Comment:
I totally appreciate you for putting in unit tests. Its definitely something that we need to do more of.
It looks like this is testing the `validateTalent` function, which should go into a `talent.test.ts` file next to `talent.ts`. This allows the unit test to test strictly what the `validateTalent` function does (which your database functions are doing in a round-about way).

Reply:
Yep, added in https://github.com/frzyc/genshin-optimizer/pull/1454/commits/0c030a3968f5570a23bae40f0fccafdffe4d69b9